### PR TITLE
Fix v0.6 depwarns

### PR DESCRIPTION
**Pull request by anriseth**
This fixes the deprecation warnings on nightly

### CODE REVIEW

**Comment by timholy** [comment]
Thanks for this. We could either leave 0.4 in the dust, or these fixes will need `Compat`. CCing @pkofod for thoughts (Optim still supports julia 0.4).

**Comment by pkofod** [comment]
Hm, I would say that we support v0.4 because it works, but I rarely do anything actively to achieve it. That said, if we can keep v0.4 support by adding some at compats I would appreciate it. I know we still have some v0.4 users who won't migrate until v 0.6 because of some regressions fixed on master that never made it into a v0.5.1 (because such a release was never made)

**Comment by anriseth** [comment]
How does a Compat fix work? Is it sufficient do 'using Compat'? (And
require Compat)

**Comment by pkofod** [comment]
If it's a hassle, don't bother, we can kill v0.4 :) But yes, you say `using Compat` and then put `@compat` in front of the deprecated/changed functionality.

**Comment by anriseth** [comment]
Hmm. Either I don't understand how to use Compat, or there is a difficulty combining `@compat` and `@test`...

**Comment by timholy** [comment]
Nice work, @anriseth!

**Comment by pkofod** [comment]
This ended up supporting v0.4 to v0.6 (inclusive) right? Maybe tag a new version then?

**Comment by timholy** [comment]
https://github.com/JuliaLang/METADATA.jl/pull/7750

**Comment by tkelman** [comment]
you rarely need `@compat` for anything that isn't a syntax transformation

File: test/runtests.jl [hunk 1]
```diff
@@ -1,53 +1,58 @@
...
     F, d = ldltfact(Positive, A, pivot)
-    @test_approx_eq full(F) A
+    @test @compat full(F) ≈ A
     @test all(d .== 1)
 end
 
 A = [1 0; 0 -2]
 F = eigfact(Positive, A)
-@test_approx_eq full(F) abs(A)
+if VERSION < v"v0.5-"
```
Comment (tkelman) [inline test/runtests.jl]: this cutoff isn't correct, not all 0.5-pre versions had `≈`

Comment (anriseth) [inline test/runtests.jl]: How about:
```jl
if isdefined(≈)                                                                 
    @test full(F) ≈ abs.(A)                                                     
                                                                                
else                                                                            
    @test_approx_eq full(F) abs(A)                                              
end   
```

Comment (anriseth) [inline test/runtests.jl]: `isdefined(:≈)`

**Pull request merged**